server/src/main/java/org/spine3/server/aggregate/Aggregate.java

```diff
@@ -542,4 +545,97 @@ private IllegalStateException missingEventApplier(Class<? extends Message> event
+     * @param newValue the new value requested in the command
+     * @return new {@code ValueMismatch} instance
+     */
+    @SuppressWarnings("unused") // part of API
+    protected ValueMismatch expectedDefault(Message actual, Message newValue) {
+        return MessageMismatch.expectedDefault(actual, newValue, getVersion());
+    }
+
+    /**
+     * Creates a {@code ValueMismatch} for a command that wanted to <em>clear</em> a value,
+     * but discovered that the field already has the default value.
+     *
+     * @param expected the value of the field that the command wanted to clear
+     * @return new {@code ValueMismatch} instance
+     */
+    @SuppressWarnings("unused") // part of API
+    protected ValueMismatch expectedNotDefault(Message expected) {
+        return MessageMismatch.expectedNotDefault(expected, getVersion());
+    }
+
+    /**
+     * Creates a {@code ValueMismatch} for a command that wanted to <em>change</em> a field value,
+     * but discovered that the field has the default value.
+     *
+     * @param expected the value expected by the command
+     * @param newValue the value the command wanted to set
+     * @return new {@code ValueMismatch} instance
+     */
+    @SuppressWarnings("unused") // part of API
+    protected ValueMismatch expectedNotDefault(Message expected, Message newValue) {
+        return MessageMismatch.expectedNotDefault(expected, newValue, getVersion());
+    }
+
+    /**
+     * Creates {@code ValueMismatch} for the case of discovering a value different than by a command.
+     *
+     * @param expected the value expected by the command
+     * @param actual   the value discovered instead of the expected value
+     * @param newValue the new value requested in the command
+     * @return new {@code ValueMismatch} instance
+     */
+    @SuppressWarnings("unused") // part of API
+    protected ValueMismatch unexpectedValue(Message expected, Message actual, Message newValue) {
+        return MessageMismatch.unexpectedValue(expected, actual, newValue, getVersion());
+    }
+
+    /**
+     * Creates {@code ValueMismatch} for the case of discovering a non-empty value,
+     * when an empty string was expected by a command.
+     *
+     * @param actual   the value discovered instead of the empty string
+     * @param newValue the new value requested in the command
+     * @return new {@code ValueMismatch} instance
+     */
+    @SuppressWarnings("unused") // part of API
+    protected ValueMismatch expectedEmpty(String actual, String newValue) {
+        return StringMismatch.expectedEmpty(actual, newValue, getVersion());
+    }
+
+    /**
+     * Creates a {@code ValueMismatch} for a command that wanted to clear a string value,
+     * but discovered that the field is already empty.
+     *
+     * @param expected the value of the field that the command wanted to clear
+     * @return new ValueMismatch instance
+     */
+    @SuppressWarnings("unused") // part of API
```

Comment:
As discussed vocally, let's cover these methods with unit tests and remove the warning suppression.